### PR TITLE
Document missing shell `_function` and `_completion` variables

### PR DESCRIPTION
**Pull request by danielbayley**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by Bo98** [comment]
Should probably do the other shells too: https://github.com/Homebrew/brew/blob/9a48d24545eac467f641be03a7c3762c241fa243/Library/Homebrew/formula.rb#L877-L915

File: docs/Formula-Cookbook.md [hunk 1]
```diff
@@ -661,25 +661,25 @@ Generally we'd rather you were specific about what files or directories need to
...
-| **`include`**         | `#{prefix}/include`                            | `/usr/local/Cellar/foo/0.1/include`               |
-| **`info`**            | `#{prefix}/share/info`                         | `/usr/local/Cellar/foo/0.1/share/info`            |
-| **`lib`**             | `#{prefix}/lib`                                | `/usr/local/Cellar/foo/0.1/lib`                   |
-| **`libexec`**         | `#{prefix}/libexec`                            | `/usr/local/Cellar/foo/0.1/libexec`               |
-| **`man`**             | `#{prefix}/share/man`                          | `/usr/local/Cellar/foo/0.1/share/man`             |
-| **`man[1-8]`**        | `#{prefix}/share/man/man[1-8]`                 | `/usr/local/Cellar/foo/0.1/share/man/man[1-8]`    |
-| **`sbin`**            | `#{prefix}/sbin`                               | `/usr/local/Cellar/foo/0.1/sbin`                  |
-| **`share`**           | `#{prefix}/share`                              | `/usr/local/Cellar/foo/0.1/share`                 |
-| **`pkgshare`**        | `#{prefix}/share/foo`                          | `/usr/local/Cellar/foo/0.1/share/foo`             |
```
Comment (MikeMcQuaid) [inline docs/Formula-Cookbook.md:679]: Did you intend to remove `pkgshare`? Can you re-add it? Also, as @Bo98 mentioned ideally could you do all of these: https://github.com/Homebrew/brew/blob/9a48d24545eac467f641be03a7c3762c241fa243/Library/Homebrew/formula.rb#L877-L915

Thanks ❤️

**Review by MikeMcQuaid — COMMENTED** [review]
Nice work so far!

File: docs/Formula-Cookbook.md (hunk 1 above)
Comment (danielbayley) [reply docs/Formula-Cookbook.md:679]: > Did you intend to remove `pkgshare`?

No, not sure what happened there… had to disable my markdown table editor package which kept interfering.

> Can you re-add it? Also, as @Bo98 mentioned ideally could you do all of these:

Done.

**Review by MikeMcQuaid — APPROVED** [review]
Thanks again @danielbayley!

**Pull request merged**